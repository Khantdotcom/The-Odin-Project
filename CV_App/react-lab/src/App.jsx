import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import {MyButton,MyInput} from './components'
function App() {
  const [count, setCount] = useState(0);
  const input_list = [
    { name_id: 'name', label_for: 'First Name', type: 'text', required: true, size: 10, className: 'text' },
    { name_id: 'email', label_for: 'Email', type: 'email', required: true, size: 15, className: 'text'},
    { name_id: '_lname', label_for: 'Last Name', type: 'text', required: true, size: 15, className: 'text' },
    { name_id: 'ph', label_for: 'Phone Number', type: 'number', required: true, size: 15, className: 'ph-num' },   
  ];
  const cv_display = []

  function Submit(){
     cv_display.append([MyInput.value])
  }
  function Cancel(){
    MyInput.value.clear
  }
  const button_list = [{name_id:'submit', function:Submit(),className:'button'},
    {name_id:'cancel',function:Cancel(),className:'button'}
  ]

  return (
    <div className='input-area'>
    <fieldset>    
      {input_list.map((input)=>(
      <MyInput 
      key={input.name_id}
      name_id={input.name_id}
      type={input.type}
      required={input.required}
      size = {input.size}
      label_for={input.label_for}
      className={input.className}/>))}
    </fieldset>
      {button_list.map((button)=>(
        <MyButton
          key = {button.name_id}
          name_id = {button.name_id}
          onClick={button.function}
          className={button.className}
        />
      ))}
    </div>
  )}

export default App