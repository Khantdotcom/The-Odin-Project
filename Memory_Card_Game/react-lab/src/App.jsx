import React from 'react';
import { useState } from 'react';
import reactLogo from './assets/react.svg';
import viteLogo from '/vite.svg';
import './App.css';

const App = () =>{
  const [heading,setHeading] = useState("Hi bro!");

  const clickHandler = () =>{
    setHeading("haha you clicked");
  };

  return (
    <>
      <h1>{heading}</h1>
      <div>
        <a href="https://vite.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <div className="card">
        <p>
          Edit <code>src/App.jsx</code> and save to test HMR
        </p>
      </div>
      <button type='button' onClick={clickHandler}>
        Click Me
      </button>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </>
  );
};

export default App;
