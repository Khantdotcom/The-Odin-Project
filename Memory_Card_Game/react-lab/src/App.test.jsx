// App.test.jsx

import { describe, it, expect } from "vitest";
import React,{useState} from 'react';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App component", () => {
  it("renders correct heading", () => {
    const {component} = render(<App />);
    expect(component).toMatchSnapshot();
  });

  it("renders haha you clicked after button click",async ()=>{
    const user = userEvent.setup();

    render(<App/>);
    const button = screen.getByRole("button",{name: "Click Me"});
    
    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/haha you clicked/i);
    
  });
});

