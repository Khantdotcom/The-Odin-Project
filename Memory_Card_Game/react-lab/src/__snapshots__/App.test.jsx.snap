// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders correct heading 1`] = `undefined`;
